jaxrs/src/main/java/org/jboss/as/jaxrs/deployment/JaxrsCdiIntegrationProcessor.java

```diff
@@ -65,10 +67,13 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
         try {
             module.getClassLoader().loadClass(CDI_INJECTOR_FACTORY_CLASS);
```

Comment:
@asoldano Can confirm but I see no reason why this line should be outside the block guarded by isWeldDeployment. (The class has no static initializer code that might not get run if it were moved.)  Moving it will help make the resteasy-cdi module be truly optional in more cases (i.e. user has no CDI apps.)

Not a big deal; this entire DUP doesn't get executed if weld is not installed.

Reply:
I'll wait for Alesio to confirm the movement then. 
Yes, you're right, the DUP is already being skipped if weld subsystem is not in the system, see https://github.com/wildfly/wildfly/pull/11802/files#diff-ce7f16e9b05ad7c3e3288b78fc98db9bR73

Reply:
@yersan @bstansberry , +1 on moving the line above withing the if clause below

Reply:
Moved